### DISCUSSION TITLE
Custom PlugIn inputs and outputs

### DISCUSSION THREAD

**Discussion by bhandaribhumin** [ionic-team/capacitor#3448] [Q&A]
I try to access [String] in to Plugin.swift passing from definitions.js.

//access into Plugin.swift
```

 @objc func echo(_ call: CAPPluginCall) {

        let value = call.getString("value") //this is only for get string I try to get array of strings call.getArray() somthing 
       // get values let scopes: [String] = ["Read","Write","ReadWrite"]
 }
```

// how to send from function 
const permissions = ["Read","Write","ReadWrite"]
echo(options: { value: permissions }): Promise<{ result: any }>;

**Comment by ikeith** [comment]
I'm not sure I understand your question. Are you having difficulty extracting the array of strings from the call in Swift? Based on your example, it looks like this should work:
```
let values = call.getArray("value", String.self)
```

**Reply by NLanese** [reply]
That does not work for me
